lib/p2p/Pool.ts

```diff
@@ -341,10 +351,39 @@ class Pool extends EventEmitter {
   }
 
   public banNode = async (nodePubKey: string): Promise<void> => {
-    const banned = await this.nodes.addReputationEvent(nodePubKey, ReputationEvent.ManualBan);
+    if (this.nodes.isBanned(nodePubKey)) {
+      throw errors.NODE_ALREADY_BANNED(nodePubKey);
+
+    } else {
+      const banned = await this.nodes.addReputationEvent(nodePubKey, ReputationEvent.ManualBan);
+
+      if (!banned) {
+        throw errors.NODE_UNKNOWN(nodePubKey);
+      }
+    }
+  }
 
-    if (!banned) {
-      throw errors.NODE_UNKNOWN(nodePubKey);
+  public unban = async (args: { nodePubKey: string, reconnect: boolean}): Promise<void> => {
+    const { nodePubKey, reconnect } = args;
+    if (this.nodes.isBanned(nodePubKey)) {
+      const unbanned = await this.nodes.addReputationEvent(nodePubKey, ReputationEvent.ManualUnban);
+      const uri = await this.repository.getNode(nodePubKey);
+
+      if (!unbanned) {
+        throw errors.NODE_UNKNOWN(nodePubKey);
+      }
+
+      if (uri) {
+        const nodeUri: UriParts & {reconnect: boolean} = {
+          reconnect,
+          nodePubKey,
+          host: uri.lastAddress.host,
+          port: uri.lastAddress.port,
+        };
+        this.nodes.emit('node.unban', nodeUri);
```

Comment:
I think instead of this block of code, you can just check `if (reconnect)` and then call `this.tryConnectNode`, which will use existing logic to attempt to connect to a peer first by the last working address and then by advertised addresses. This would be instead of emitting the event. I'd lean towards specifying `retryConnecting` to be true for this.